### PR TITLE
feat: Add `rebaseConflictedPrs` configuration

### LINKED ISSUE #4184: ONLY rebase a branch if requested
**What would you like Renovate to be able to do?**
Currently Renovate is rebasing whenever there is a merge conflict with master. This is good behaviour, however unfortunately it is costing us a lot of CI cycles. 

There are currently a lot of big upgrades in our repo which are sitting idle with nobody working on them yet, so we don't need to keep them up to date with master. 

Hence I think it would make sense if you can entirely disable the auto-rebasing, so that now it only rebases if you tick the checkbox or apply a label. It would not rebase even if there are merge conflicts with master.

**Describe the solution you'd like**
A configuration option that disables auto-rebasing entirely, not sure what the name would be. Also it would be super helpful if it could be based on some sort of timer, e.g. if the PR is older than 1 week then do not auto-rebase anymore. I'm not sure if the timing condition is currently supported.

**Describe alternatives you've considered**
~~I can't think of any any alternatives to get this behaviour~~
Actually we could get around it by enabling that Renovate opens PRs only by approval from the master issue... but it doesn't really solve this problem, just avoids it by adding extra steps

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by rakeshtembhurne**
`rebaseConflictedPrs` is used to enable or disable auto-rebase
in case of merge conflicts with the master branch. It is `true`
by default which means branches will be rebased if there are conflicts
in a PR. By setting it to `false`, Renovate no longer will rebase it
with the master branch if there are merge conflicts.

Closes #4184

### CODE REVIEW

File: lib/workers/branch/index.js [hunk 1]
```diff
@@ -44,6 +44,16 @@ async function processBranch(branchConfig, prHourlyLimitReached, packageFiles) {
     config.rebaseRequested = rebaseCheck(config, branchPr);
     logger.debug(`Branch pr rebase requested: ${config.rebaseRequested}`);
   }
+  if (
+    config.rebaseRequested &&
+    !config.rebaseConflictedPrs &&
```
Comment (rarkins) [inline lib/workers/branch/index.js]: This is not relevant here

**Comment by rarkins** [comment]
We want to stop auto rebase conflicted and not to block manual rebase requests

**Comment by rarkins** [comment]
@rakeshtembhurne you want to insert logic into this block: https://github.com/renovatebot/renovate/blob/4a81c20c9a65acda68ce8335e5b354747fc7b5ec/lib/workers/branch/parent.js#L63-L73

File: lib/workers/branch/index.js (hunk 1 above)
Comment (rakeshtembhurne) [reply lib/workers/branch/index.js]: Removed these changes and added changes to the appropriate file.

**Comment by rarkins** [comment]
@rakeshtembhurne I resolved the merge conflicts but you'll need to regenerate the schema JSON locally and push the change

**Review by viceice — APPROVED** [review]
LGTM

File: lib/config/definitions.js [hunk 2]
```diff
@@ -862,6 +862,12 @@ const options = [
     type: 'boolean',
     default: false,
   },
+  {
+    name: 'rebaseConflictedPrs',
+    description: 'Auto-rebase when there is conflict in PRs (GitHub only)',
```
Comment (rarkins) [inline lib/config/definitions.js]: Remove the “GitHub only” comment

File: lib/workers/branch/parent.js [hunk 3]
```diff
@@ -62,6 +62,13 @@ async function getParentBranch(config) {
   // Now check if PR is unmergeable. If so then we also rebase
   if (pr && pr.isConflicted) {
     logger.debug('PR is conflicted');
+
+    if (pr.canRebase && !config.rebaseConflictedPrs) {
```
Comment (rarkins) [inline lib/workers/branch/parent.js]: This logic should exist in the existing if pr.canRebase block below it

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/config/definitions.js (hunk 2 above)
Comment (rakeshtembhurne) [reply lib/config/definitions.js]: Done the changes

File: lib/workers/branch/parent.js (hunk 3 above)
Comment (rakeshtembhurne) [reply lib/workers/branch/parent.js]: Moved the logic to below block.

**Pull request merged**